core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -375,15 +379,50 @@ final case class App(id: AppId,
                      descriptorPath: Option[Uri],
                      extraArgs: List[String]
 ) {
-  def getProxyUrls(project: GoogleProject, proxyUrlBase: String): Map[ServiceName, URL] =
+
+  def getProxyUrls(cloudContext: CloudContext,
+                   workspaceId: Option[WorkspaceId],
+                   proxyUrlBase: String,
+                   apiVersion: String
+  ): Map[ServiceName, URL] =
     appResources.services.map { service =>
```

Comment:
The logic here is kind of tricky actually. :) 

Thinking through the cases, I think we want to generate v1 or v2 style URLs based on the API being called. But we probably shouldn't fail if there are apps in the system that I can't generate a URL for -- e.g. if I call `listAppsV2` and I have some v1 apps in the system.

I came up with this (warning: didn't compile!). Does this cover all the cases? Btw I like using a pattern match to describe this type of thing.

```
appResources.services.flatMap { service =>
  val proxyPathOpt = (apiVersion, cloudContext, workspaceId) match {
    // v1 apis should always generate urls for google apps
    case ("v1", CloudContext.Gcp(project), _) =>  
      Some(s"google/v1/apps/${project.value}/${appName.value}/${kubernetesService.config.name.value}")
    // v2 apis should generate urls for apps on both clouds, but only if there's a workspace
    case ("v2", _, Some(workspace)) => 
      Some(s"apps/v2/${workspaceId.value}/${appName.value}/${kubernetesService.config.name.value}")
    // otherwise don't generate a proxy url
    case _ =>
      None
  }
  proxyPathOpt match {
    case Some(proxyPath) =>
      Map(kubernetesService.config.name -> new URL(s"${proxyUrlBase}${proxyPath}${kubernetesService.config.path.getOrElse("")}"))
    case None =>
      Map.empty
  }
}
```

Reply:
This looks great! I fixed up the errors and I like it a lot more than my first solution.
Mostly did it as a first pass, so I appreciate the follow up.